simtools/data_model/validate_data.py

```diff
@@ -293,6 +294,33 @@ def _get_reference_unit(self, column_name):
 
         return u.Unit(reference_unit)
 
+    def _check_data_type(self, col, column_name):
+        """
+        Check column data type.
+
+        Parameters
+        ----------
+        col: astropy.column or Quantity
+            data column to be converted
+        column_name: str
+            column name
+
+        Raises
+        ------
+        TypeError
+            if data type is not correct
+
+        """
+
+        reference_dtype = self._get_reference_data_column(column_name).get("type", None)
+
+        if not np.issubdtype(col.dtype, reference_dtype):
+            self._logger.error(
+                f"Invalid data type in column '{column_name}'. "
+                f"Expected type '{reference_dtype}', found '{col.dtype}'"
+            )
+            raise TypeError
+
```

Comment:
I think we could include either a debug or a info logger message here to inform the user that the column is validated.

Reply:
and then test in the unit tests for a successful verification of the column catching the logger message

Reply:
Added.